typescript/xstate/src/lib.ts

```diff
@@ -0,0 +1,360 @@
+    _register: (sessionId, actorRef) => {
+      if (actorRef.id in childrenByID) {
+        // rehydration case; ensure session ID maintains continuity
+        sessionId = childrenByID[actorRef.id].sessionId
+        actorRef.sessionId = sessionId
+      } else {
+        // new actor case
+        childrenByID[actorRef.id] = serialiseActorRef(actorRef)
+        ctx.set("children", childrenByID)
+      }
+      console.log("register", sessionId, actorRef.id)
+      children.set(sessionId, actorRef);
+      return sessionId;
+    },
+    _unregister: (actorRef) => {
+      if (actorRef.id in childrenByID) {
+        // rehydration case; ensure session ID maintains continuity
+        actorRef.sessionId = childrenByID[actorRef.id].sessionId
+      }
+
+      children.delete(actorRef.sessionId);
+      delete childrenByID[actorRef.id]
+      ctx.set("children", childrenByID)
+      const systemId = reverseKeyedActors.get(actorRef);
+
+      if (systemId !== undefined) {
+        keyedActors.delete(systemId);
+        reverseKeyedActors.delete(actorRef);
+      }
+    },
+    _sendInspectionEvent: (event) => {
+      const resolvedInspectionEvent: InspectionEvent = {
+        ...event,
+        rootId: "root",
+      };
+      observers.forEach((observer) => observer.next?.(resolvedInspectionEvent));
+    },
+    actor: (sessionId) => {
+      return children.get(sessionId)
+    },
+    get: (systemId) => {
+      return keyedActors.get(systemId) as T['actors'][any];
+    },
+    _set: (systemId, actorRef) => {
+      const existing = keyedActors.get(systemId);
+      if (existing && existing !== actorRef) {
+        throw new Error(
+          `Actor with system ID '${systemId as string}' already exists.`
+        );
+      }
+
+      keyedActors.set(systemId, actorRef);
+      reverseKeyedActors.set(actorRef, systemId);
+    },
+    inspect: (observer) => {
+      observers.add(observer);
+    },
+    _relay: (source, target, event) => {
+      console.log("Relaying message from", source?.id, "to", target.id, ":", event.type);
+      (target as any)._send(event);
+    },
+    scheduler,
+    getSnapshot: () => {
+      return {
+        _scheduledEvents: {} // unused
+      };
+    },
+    start: () => {
+    }
+  };
+
+  return system;
+}
+
+interface FakeParent<TLogic extends AnyStateMachine> extends AnyActorRef {
+  _send: (event: EventFromLogic<TLogic>) => void;
+}
+
+export async function createActor<TLogic extends AnyStateMachine>(ctx: restate.RpcContext, api: XStateApi<TLogic>, systemName: string, logic: TLogic extends AnyStateMachine ? AreAllImplementationsAssumedToBeProvided<TLogic['__TResolvedTypesMeta']> extends true ? TLogic : MissingImplementationsError<TLogic['__TResolvedTypesMeta']> : TLogic, options?: ActorOptions<TLogic>): Promise<Actor<TLogic>> {
+  const system = await createSystem(ctx, api, systemName)
+  const snapshot = await ctx.get<Snapshot<unknown>>("snapshot") ?? undefined;
+
+  const parent: FakeParent<TLogic> = {
```

Comment:
hack; I can't find a way to create an actor with a particular system set, so I had to create a 'fake' parent for the new actor which has a system already set